File: 717-1BitAnd2BitCharacters/717-1BitAnd2BitCharacters.py
# Last updated: 12/6/2025, 5:47:28 am
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        prev = 0

        for i in range(n-1):
            if bits[i]:
                prev = not prev
            else:
                prev = 0
        
        return not prev

File: 717-1-bitand2-bitCharacters/717-1-bitand2-bitCharacters.cpp
// Last updated: 18/11/2025, 10:01:44 am
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        int n = bits.size();
        int i = 0;
        while (i < n-1) {
            if (bits[i] == 1) i++;
            i++;
        }
        if (i == n-1) return true;
        return false;
    }
};